src/main/java/com/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -25,10 +24,10 @@ public AuthenticationServiceImpl(UserService userService,
 
     @Override
     public User login(String email, String password) throws AuthenticationException {
-        Optional<User> userFromDb = userService.findByEmail(email);
-        if (userFromDb.isPresent() && userFromDb.get().getPassword()
-                .equals(HashUtil.hashPassword(password, userFromDb.get().getSalt()))) {
-            return userFromDb.get();
+        User userFromDb = userService.findByEmail(email);
+        if (userFromDb != null && userFromDb.getPassword()
```

Comment:
first - can it be null?

second - the problem is that earlier, when your user entered incorrect email, the AuthenticationException was thrown with the informative message

and now the NoSuchElementException is thrown

my advice is not to change `findByEmail()` method
but I think you won't like calling optional.get() all the time in your controllers, so let's create `getById(Long userId)` and pass userId as the `@RequestParam` to the controllers?
what do you think?